modules/web-test/variables.tf

```diff
@@ -0,0 +1,115 @@
+variable "resource_group_name" {
+  description = "The name of the resource group to create the resources in."
+  type        = string
+}
+
+variable "location" {
+  description = "The location to create the resources in."
+  type        = string
+}
+
+variable "component_id" {
+  description = "The Application Insights component ID."
+  type        = string
+}
+
+variable "name" {
+  description = "The name of this Web Test."
+  type        = string
+}
+
+variable "configuration" {
+  description = "An XML configuration specification for this Web Test."
+  type        = string
+  default     = null
+}
+
+variable "kind" {
+  description = "The kind of web test that this Web Test watches."
+  type        = string
+  default     = "standard"
+
+  validation {
+    condition     = contains(["standard", "ping", "multistep"], var.kind)
+    error_message = "Web Test kind must be \"standard\", \"ping\" or \"multistep\"."
+  }
+}
+
+variable "description" {
+  description = "The purpose/user defined description of this Web Test."
+  type        = string
+  default     = null
+}
+
+variable "enabled" {
+  description = "Is this Web Test enabled?"
+  type        = bool
+  default     = false
+}
+
+variable "frequency" {
+  description = "Interval in seconds between test runs for this Web Test."
+  type        = number
+  default     = 300
+}
+
+variable "retry_enabled" {
+  description = "Is retry on Web Test failure enabled?"
+  type        = bool
+  default     = false
+}
+
+variable "timeout" {
+  description = "Seconds until this Web Test will timeout and fail."
+  type        = number
+  default     = 30
+}
+
+variable "geo_locations" {
+  description = "A list of physical locations to run this Web Test."
+  type        = list(string)
+  default     = ["emea-gb-db3-azr"] # North Europe
```

Comment:
Not sure it's a good idea to have a default value here. I think location is something that should always be set explicitly by the user.

Reply:
Agreed, should default then be completely omitted or an empty list ( [ ] )?

Reply:
It depends. I see in docs that it's marked as required, so I guess the best option would be to omit it. Could test if it's possible with an empty list though, but I doubt it.

Reply:
I'll investigate 👍🏼

Reply:
I'll omit the default argument for now.